From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: acrylic-style <me@acrylicstyle.xyz>
Date: Tue, 9 Nov 2021 20:52:30 +0900
Subject: [PATCH] Enforce max-level of job


diff --git a/src/main/java/com/gamingmesh/jobs/container/JobProgression.java b/src/main/java/com/gamingmesh/jobs/container/JobProgression.java
index 354566f06043967a5b5dd44e74fd9d04f811f3b7..9527a7d30f0be92f168b5fb227107a93956f823c 100644
--- a/src/main/java/com/gamingmesh/jobs/container/JobProgression.java
+++ b/src/main/java/com/gamingmesh/jobs/container/JobProgression.java
@@ -133,7 +133,7 @@ public class JobProgression {
      * @return the level of this job
      */
     public int getLevel() {
-	return level;
+	return Math.min(level, job.getMaxLevel());
     }
 
     /**
